### DISCUSSION TITLE
Importing objects with custom fields specified by name instead of id?

### DISCUSSION THREAD

**Discussion by dpotter-intoto** [netbox-community/netbox#12807] [Q&A]
I'd like to import a bunch of `ipam.prefix` objects, but my netbox data model for a prefix includes an optional custom field to a `dcim.device` object named **cf_associated_switch**.

Is it possible for me to perform a file import (CSV/JSON/YAML) of these prefix objects with the **cf_associated_switch** field specified by device name instead of by device id?

**Comment by jordanrvillarreal** [comment]
It should be possible to do with CSV but currently not with JSON/YAML.  Try:

`cf_associated_switch.name`
